lib/rc_test.sh

```diff
@@ -0,0 +1,276 @@
+#!/usr/bin/env bash
+
+# mock `pulumi` binary
+#
+# This prevents the actual `pulumi` binary from being executed during
+# these tests. Every invocation is logged to a file.
+#
+# To make assertions, simply inspect the contents of the file to
+# ensure that the expected commands _would_ have been invoked.
+#
+# In the case of the `pulumi config get artifacts`, we actually do
+# need to simulate a functioning binary. In that case, the behavior is
+# governed by the $EXISTING_ARTIFACTS environment variable, which may
+# be set in a test. If it is not present, the command errors out, as
+# it would in "real life". If it is present, its value is returned.
+pulumi() {
+    echo "${FUNCNAME[0]} $*" >> "${ALL_COMMANDS}"
+
+    case "$*" in
+        config\ get\ artifacts*)
+            # Expects EXISTING_ARTIFACTS to be a valid, non-empty JSON object
+            if [ -n "${EXISTING_ARTIFACTS:-}" ]; then
+                echo "${EXISTING_ARTIFACTS}"
+            else
+                # Simulates the error condition where the expected
+                # config key was not present
+                return 255
+            fi
+            ;;
+        *) ;;
+    esac
+}
+
+# Simple git binary mock that just records its invocations
+git() {
+    echo "${FUNCNAME[0]} $*" >> "${ALL_COMMANDS}"
+}
+
+recorded_commands() {
+    if [ -f "${ALL_COMMANDS}" ]; then
+        cat "${ALL_COMMANDS}"
+    fi
+}
+
+oneTimeSetUp() {
+    export BUILDKITE_BUILD_URL="https://buildkite.com/grapl/pipeline-infrastructure-verify/builds/2112"
+    export ALL_COMMANDS="${SHUNIT_TMPDIR}/all_commands"
+    export GIT_AUTHOR_NAME="Testy McTestface"
+    export GIT_AUTHOR_EMAIL="tests@tests.com"
```

Comment:
I'm assuming this won't come up or be an issue, but `tests.com` is a real site. Maybe we should use `example.com` here?

Reply:
Ah, good point. It won't be an issue here, but it's probably not a bad habit to start using `example.com` more broadly.